scala/lang/security/audit/tainted-sql-string.yaml

```diff
@@ -1,68 +1,83 @@
 rules:
-  - id: tainted-sql-string
-    languages: [scala]
-    severity: ERROR
-    mode: taint
-    message: >-
-      User data flows into this manually-constructed SQL string. User data
-      can be safely inserted into SQL strings using prepared statements or an
-      object-relational mapper (ORM). Manually-constructed SQL strings is a
-      possible indicator of SQL injection, which could let an attacker steal or
-      manipulate data from the database. Instead, use prepared statements
-      (`connection.PreparedStatement`) or a safe library.
-    metadata:
-      cwe: "CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')"
-      owasp:
-        - A03:2021
-        - A01:2017
-      references:
-        - https://docs.oracle.com/javase/7/docs/api/java/sql/PreparedStatement.html
-      category: security
-      technology:
+- id: tainted-sql-string
+  languages:
+    - scala
+  severity: ERROR
+  mode: taint
+  message: User data flows into this manually-constructed SQL string. User data
+    can be safely inserted into SQL strings using prepared statements or an
+    object-relational mapper (ORM). Manually-constructed SQL strings is a possible
+    indicator of SQL injection, which could let an attacker steal or manipulate
+    data from the database. Instead, use prepared statements
+    (`connection.PreparedStatement`) or a safe library.
+  metadata:
+    cwe: "CWE-89: Improper Neutralization of Special Elements used in an SQL Command
+      ('SQL Injection')"
+    owasp:
+      - A03:2021
+      - A01:2017
+    references:
+      - https://docs.oracle.com/javase/7/docs/api/java/sql/PreparedStatement.html
+    category: security
+    technology:
       - scala
-      confidence: MEDIUM
-    pattern-sources:
+    confidence: MEDIUM
+    license: Commons Clause License Condition v1.0[LGPL-2.1-only]
+  pattern-sources:
     - patterns:
-      - pattern: $PARAM
-      - pattern-either:
-        - pattern-inside: |
-            def $CTRL(..., $PARAM: $TYPE, ...) = {
-              ...
-            }
-        - pattern-inside: |
-            def $CTRL(..., $PARAM: $TYPE, ...) = $A {
-              ...
-            }
-        - pattern-inside: |
-            def $CTRL(..., $PARAM: $TYPE, ...) = $A(...) {
-              ...
-            }
-    pattern-sinks:
+        - pattern: $PARAM
+        - pattern-either:
+            - pattern-inside: |
+                def $CTRL(..., $PARAM: $TYPE, ...) = {
+                  ...
+                }
+            - pattern-inside: |
+                def $CTRL(..., $PARAM: $TYPE, ...) = $A {
+                  ...
+                }
+            - pattern-inside: |
+                def $CTRL(..., $PARAM: $TYPE, ...) = $A(...) {
+                  ...
+                }
+  pattern-sinks:
```

Comment:
This isn't really a "SQL injection" rule as much as it's a "string concatenation" rule with some additional keywords. As a fix, this is probably fine but I think we'll get better results by scrapping this approach and directly targeting Java/Scala ORMs as sinks.